flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosTaskManagerParameters.java

```diff
@@ -422,6 +422,29 @@ private static double getCpuCores(final Configuration configuration) {
 		return TaskExecutorResourceUtils.getCpuCoresWithFallback(configuration, fallback).getValue().doubleValue();
 	}
 
+	private static MemorySize getTotalProcessMemory(final Configuration configuration) {
+		MemorySize legacyTotalProcessMemory = MemorySize.parse(configuration.getInteger(MESOS_RM_TASKS_MEMORY_MB) + "m");
+		MemorySize unifiedTotalProcessMemory = MemorySize.parse(configuration.getString(TaskManagerOptions.TOTAL_PROCESS_MEMORY, "0"));
```

Comment:
What should happen if neither of the two options are set? At the moment we will default to the default value of `MESOS_RM_TASKS_MEMORY_MB`. Is this what we want in the long run or should it rather fail because we want the user to configure `TOTAL_PROCESS_MEMORY`? The implication would be to only consider `MESOS_RM_TASKS_MEMORY_MB` if it has been explicitly set.

Reply:
IMO, from the perspective of backwards compatibility, if the new unified `TOTAL_PROCESS_MEMORY` is not explicitly configured, we should fallback to the legacy `MESOS_RM_TASKS_MEMORY_MB`, including fallback to its default value if not explicitly configured.

Reply:
I agree with that. This way we avoid failing legacy jobs which did not configure this option at all but relied on its default value. I have also added a commit to deprecate `MESOS_RM_TASKS_MEMORY_MB`. Once we remove this deprecated option (https://issues.apache.org/jira/browse/FLINK-15198), we can align the behaviour with FLIP-49 where we require to set it explicitly.

Reply:
All right. Thanks for the clarification. Sounds good to me.